static/app/components/nav/secondarySidebar.tsx

```diff
@@ -0,0 +1,30 @@
+import {useEffect, useRef} from 'react';
+import styled from '@emotion/styled';
+
+import {useNavContext} from 'sentry/components/nav/context';
+
+export function SecondarySidebar() {
+  const ref = useRef<HTMLDivElement>(null);
+  const {setSecondaryNavEl} = useNavContext();
+
+  useEffect(() => {
+    if (ref.current) {
+      setSecondaryNavEl(ref.current);
+    }
+  }, [ref, setSecondaryNavEl]);
+
+  return <SecondarySidebarWrapper ref={ref} />;
```

Comment:
i don't think refs trigger useEffects. You  might be seeing the useMemo from the context trigger the useEffect at the right time. This will also set it to null on unmount
```suggestion
  const {setSecondaryNavEl} = useNavContext();

  return <SecondarySidebarWrapper ref={setSecondaryNavEl} />;
```

Reply:
Oh good call, this is much better